assets/pages/regular-contributions/regularContributions.jsx

```diff
@@ -69,38 +56,39 @@ const state: PageState = store.getState();
 
 // ----- Render ----- //
 
-const content = (
-  <Provider store={store}>
-    <div className="gu-content">
-      <TestUserBanner />
-      <SimpleHeader />
-      <CirclesIntroduction headings={title[contributionType.toLowerCase()]} />
-      <hr className="regular-contrib__multiline" />
-      <div className="regular-contrib gu-content-margin">
-        <InfoSection heading={`Your ${contributionType.toLowerCase()} contribution`} className="regular-contrib__your-contrib">
-          <PaymentAmount
-            amount={state.page.regularContrib.amount}
-            currency={state.page.regularContrib.currency}
-          />
-          <Secure />
-        </InfoSection>
-        <InfoSection heading="Your details" headingContent={<Signout />} className="regular-contrib__your-details">
-          <DisplayName />
-          <FormFields />
-        </InfoSection>
-        <InfoSection heading="Payment methods" className="regular-contrib__payment-methods">
-          <RegularContributionsPayment contributionType={contributionType} />
-        </InfoSection>
-      </div>
-      <div className="terms-privacy gu-content-filler">
-        <InfoSection className="terms-privacy__content gu-content-filler__inner">
-          <TermsPrivacy country={country} />
-          <ContribLegal />
-        </InfoSection>
+const CheckoutPage = () => (
+  <RegularContributionsPage
+    amount={state.page.regularContrib.amount}
+    currency={state.page.regularContrib.currency}
+    contributionType={contributionType}
+    country={country}
+  />
+);
+
+const ThankYouPage = () => {
+  const current: PageState = store.getState();
+  if (current.page.regularContrib.paymentMethod === 'directDebit') {
+    return (
+      <DirectDebitContributionsThankYouPage
+        accountHolderName={current.page.directDebit.accountHolderName}
+        accountNumber={current.page.directDebit.accountNumber}
+        sortCodeArray={current.page.directDebit.sortCodeArray}
+      />
+    );
+  }
+  return (<DefaultContributionsThankYouPage />);
+};
+
+const router = (
+  <BrowserRouter>
+    <Provider store={store}>
+      <div>
+        <Route exact path="/contribute/recurring" component={CheckoutPage} />
+        <Route exact path="/contribute/recurring/thankyou" component={ThankYouPage} />
+        <Route exact path="/contribute/recurring/pending" component={ThankYouPage} />
       </div>
-      <Footer />
-    </div>
-  </Provider>
+    </Provider>
```

Comment:
The way this is currently set up, these pages aren't `connect`ed. Assuming the `Route` components obey normal React behaviour, the components that live within them (i.e. the `component` prop) can have access to the Redux store through the `Provider`. Rather than doing `store.getState()` to obtain a static state object and pass it into the "page" components, could we `connect` these components to the store in their own files? Then we can pass the `connect`ed components into the `Route` component.

Reply:
That is probably true.  I was trying to avoid changing any more than necessary, and store.getState() was already used in this file.  I think

Reply:
I've connected these now.

Reply:
👍 🎉